src/app/modules/signuplogin/components/editprofile.component.ts

```diff
@@ -0,0 +1,120 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { SignUpViewModel } from './loginsignup.component';
+import {DataService} from '../../../shared/services/data-service.service';
+import {HttpClient,HttpParams} from '@angular/common/http';
+import {Router} from '@angular/router';
+import {MatChipInputEvent} from '@angular/material/chips';
+import {COMMA, ENTER} from '@angular/cdk/keycodes';
+import {MatIconModule} from '@angular/material/icon';
+
+@Component({
+  selector: 'app-editprofile',
+  templateUrl: '../pages/editprofile.component.html',
+  styleUrls: ['../css/editprofile.component.css']
+})
+export class EditprofileComponent implements OnInit {
+
+  email_Id:string;
+  response:any;
+  arrays:object[];
+  url1:any;
+  linkedInUrl:any;
+
+//Tagging code
+    visible = true;
+    selectable = true;
+    removable = true;
+    addOnBlur = true;
+    selectedFile:any;
+    readonly separatorKeysCodes: number[] = [ENTER, COMMA];
+    technologies: Technology[] = [
+    {name: 'Java'},
+    {name: 'PHP'},
+    {name: 'Spring'},
+  ];
+
+  eProfileModel:EditProfileModel={
+    emailid:'',
+    technologies:'',
+    linkedInUrl:''
+  };
+
+   viewModel:SignUpViewModel;
+
+  constructor(private transferService:DataService,private http:HttpClient,
+    private router: Router){
+    this.email_Id = transferService.getData();
+  }
+
+  ngOnInit(): void {
+
+  }
+
+  onSelectFile(event) {
+      if (event.target.files && event.target.files[0]) {
+        var reader = new FileReader();
+        this.selectedFile=event.target.files[0];
+        reader.readAsDataURL(event.target.files[0]); // read file as data url
+
+        reader.onload = (event) => { // called once readAsDataURL is completed
+          this.url1 = event.target.result;
+        }
+      }
+    }
+
+
+ submit():void{
+   this.eProfileModel.emailid = this.email_Id;
+   this.eProfileModel.technologies=JSON.stringify(this.technologies);
+   var uploadData=new FormData();
+
+   if(this.selectedFile!=null)
+   {
+   uploadData.append('myFile',this.selectedFile,this.selectedFile.name);
+   }  
+   var value=JSON.stringify( this.eProfileModel);
+   uploadData.append('model',value);
+   let url = "http://localhost:8787/api/updateProfile";
+   this.http.patch(url,uploadData).subscribe(
```

Comment:
please convert the http call through httpService that is created. Refer to examples in signup.
Write a method in Httpservice, try to make this method as generic as possible, so that others can reuse it.